src/core/server/http/cookie_session_storage.ts

```diff
@@ -71,8 +73,9 @@ export async function createCookieSessionStorageFactory<T>(
   });
 
   return {
-    asScoped(request: Request) {
-      return new ScopedCookieSessionStorage<T>(server, request);
+    asScoped(request: Readonly<Request> | KibanaRequest) {
```

Comment:
As soon as https://github.com/elastic/kibana/pull/36509 is implemented we pass bound session storage to a consumer. There is an open question on how to restrict access to this API outside of the plugin boundaries.
https://github.com/elastic/kibana/pull/36509#discussion_r290186281